api/src/main/java/org/openmrs/module/reporting/report/service/db/HibernateReportDAO.java

```diff
@@ -214,7 +216,44 @@ public List<ReportRequest> getReportRequests(ReportDefinition reportDefinition,
 		}
 		return c.list();
 	}
-	
+
+	/**
+	 * @see ReportDAO#getReportsWithPagination(ReportDefinition, Date, Date, Integer, Integer, Status...)
+	 */
+	public ReportRequestDTO getReportsWithPagination(ReportDefinition reportDefinition, Date requestOnOrAfter, Date requestOnOrBefore, Integer pageNumber, Integer pageSize, Status...statuses) {
```

Comment:
Since you are returning requests, `getReportRequests` looks better than `getReports`. Not so? 😊

Reply:
Changed to `getReportRequestsWithPagination`, although the suffix `WithPagination` might not be great.